### DISCUSSION TITLE
Driving the LEGO® Technic Liebherr Crawler Crane LR 13000 (42146) with the Powered Up remote

### DISCUSSION THREAD

**Discussion by stevo1957** [pybricks/support#1985] [Q&A]
I have followed the guide and installed Pybricks to the hubs, but can not get the remote to work.

I had issues installing the firmware and code from my laptop (connection issues - it wouldn't connect) but was able to install firmware when I used my Android phone.

I downloaded the required code to the upper and lower hubs but when I turn the lego remote unit on, it does not connect. 

Just wondering what I need to do to get this to work.

Appreciate any help.

**Comment by laurensvalk** [comment]
Do you see any errors in the pybricks code window output at the bottom? What light (or blink) pattern do you see on the hub? 

Could you start with a smaller program that does not involve two hubs at once? This can help you get a bit more familiar with Pybricks.

**Comment by stevo1957** [comment]
Hi Laurens,

Thank you so much for your quick reply.

I think my issue may be that I have not unlocked the code blocks yet. I am thinking of getting a monthly subscription through Patreon.

In the meantime, I'd like to try the program to see if it is what I am looking for, so have now followed the second part of your instruction and compiled a Python program file.

I am getting the following error though:

Failed to compile MicroPython file.
Traceback (most recent call last):
  File "Crane.py", line 24
IndentationError: unexpected indent

I just used the copy code button in your instruction and paste at line 1 in a new blank python file which I have named "crane".

Regards,

Steve Jones
________________________________

**Reply by BertLindeman** [reply]
Hello,

Maybe Laurens suggested to run a small program to prove the remote does connect.

Something like this maybe?
```python
from pybricks.hubs import TechnicHub
from pybricks.pupdevices import Remote
from pybricks.tools import wait
from pybricks.parameters import Color

hub = TechnicHub()
remote = Remote(timeout=None)

for i in range(10):
    hub.light.on(Color.RED)
    remote.light.on(Color.RED)
    wait(50)
    remote.light.on(Color.GREEN)
    hub.light.on(Color.GREEN)
    wait(50)
```

Run this program and press the center button on the remote.
My result is a short blinking of the hub and remote light RED / GREEN.

Bert

**Reply by stevo1957** [reply]
Thanks for the file Bert.
I have been able to get things working. I used another computer with a bluetooth dongle and successfully connected to the hubs. 
I can download the 2 python code files, hit the run button in Pybricks and then turn on the remote and it works fine.
What I am trying to do, is establish the direct connection so I don't have to use Pybricks each time on the PC. 
Despite researching this, I'm having trouble getting it to work.

Am I able to do this using the Python code programming or do I need to use block programming?

Thanks for your help.

**Reply by BertLindeman** [reply]
That **is** good news. 

1. Your last question first:
The block programs are the same as the python programs, difference is **only** that the block programs have the blocks.
To run the programs you can use the python code. 

2. . You download the hub programs to the hub (with the `Run` button).
 .  The hub remembers the program
. To start the programs press the green button on the hub to put them on (if they are off)
. Start the program by pressing the green hub button. 

That is all. No PC or tablet needed anymore.

Hope this helps.

Bert

**Comment by stevo1957** [comment]
Thanks Bert,

I still can't get the hubs to remember the code so that I don't need the PC on with Pybricks.
I don't know what I am doing wrong.
Remote will work providing I don't turn off Pybricks after I run the programs.

Here's what I have done:


  1.
Turn on both hubs - double blue flashing lights
  2.
Connect crane hub with Pybricks via the bluetooth button
  3.
Run the crane program via the Pybricks run button - now slow single blue flashing light
  4.
Disconnect the bluetooth to the crane hub using the bluetooth button
  5.
Connect base hub with Pybricks via the bluetooth button
  6.
Run the base program via the Pybricks run button - now slow single blue flashing light
  7.
Disconnect the bluetooth to the base hub using the bluetooth button
  8.
Turn off both hubs
  9.
Exit Pybricks
  10.
Turn on both hubs which double flash blue
  11.
Turn on remote which double flashes white
  12.
Nothing connects

Any ideas?

Regards,
Steve
________________________________

**Reply by BertLindeman** [reply]
Almost there! Steve.

11. Turn on remote which double flashes white
12. press the green button once more on both hubs (they are already double flash blue) to start the program.

11 and 12 maybe in any order but close to eachother.

That should get you running. 

Bert

**Reply by BertLindeman** [reply]
Ah, Steve,

I do not mean to say "re-do step 1 through 10"
All you need to do in this stage is:
1. press both hubs on (double flash blue)
2. Turn on remote which double flashes white
3. press the green button once more on both hubs (they are already double flash blue) to start the program.

**Comment by stevo1957** [comment]
Thanks for your help with this Bert.

I'm having issues with the base hub......it won't go to a slow single blue flash when I run the program.

I'll reinstall the code on both hubs again and see how things go.

I'll report back later!

Cheers,
Steve

________________________________

**Comment by stevo1957** [comment]
Thanks Bert, I'm only doing the extra steps you gave me, but there is something wrong with the base hub....it won't let me reload the code.

I'll flash back to lego firmware and then reinstall Pybricks and try again.

Cheers,

Steve
________________________________

**Reply by stevo1957** [reply]
Actually, I just need to reinstall Pybricks only...Inote it erases old Pybricks firmware and reinstalls again.

**Comment by stevo1957** [comment]
Success thanks Bert!

I need to:
Turn on remote
Turn on top hub
Turn on bottom hub
Press button to run program on top hub
Press button to run program on bottom hub

Everything then connects and it works well.

I have really appreciated your help with this Bert. I know nothing about Pybricks but hope to learn.

Regards,

Steve

________________________________

**Reply by BertLindeman** [reply]
Hurray! 👍 

Well done Steve. Your detailed list of steps made it easy to help you!
  
You already learned a solid base knowledge.

Have fun.

Bert

**Reply by laurensvalk** [reply]
Thanks so much @BertLindeman for helping out! Another happy user.

@stevo1957 -- here's a table with the meaning of the light indicators: https://pybricks.com/learn/getting-started/pybricks-environment/#start-programs-with-the-hub-button And how you can run programs again without the computer involved.

You have figured this out already, but this may help others reading along here.

For your next project, perhaps use the parts from the crane to build [something like this](https://pybricks.com/learn/building-a-robot/technic/), and gradually explore all the things you can do, build, and learn :)

**Reply by stevo1957** [reply]
Thanks for the links, Laurens. 

Everything works fine now except that I have to use a different computer to connect to the hubs. For some reason, my Acer laptop will see the hub but will give connection errors when I try and connect. I've tried updating drivers but problem still persists. Good thing is everything works fine on my desktop PC.

Thanks again to you and @BertLindeman for your help. My grandkids will love using the remote to drive the crane around.

**Reply by tstr14** [reply]
Please help me understand this:

Success thanks Bert!

I need to:
Turn on remote  **OK**
Turn on top hub **OK**
Turn on bottom hub **OK**
Press button to run program on top hub   **Press which button? The green one on the Hub?**
Press button to run program on bottom hub   **Same question.**

Everything then connects and it works well.

I have really appreciated your help with this Bert. I know nothing about Pybricks but hope to learn.

Regards,

Steve

**Reply by laurensvalk** [reply]
The key is that both hubs should be:
1. On
2. Running a program.

For step 1, simply press the green button.

For step 2, start the program with the app. After that, you can just start it again with the green button.

So that makes it seem like "two presses", one for turning on the hub and one for starting the program you put on it earlier.
